pkg/network/state.go

```diff
@@ -918,7 +920,8 @@ func (ns *networkState) determineConnectionIntraHost(connections slice.Chain[Con
 			_, conn.IntraHost = lAddrs[keyWithRAddr]
 		}
 
-		fixConnectionDirection(conn)
+		fixOutgoingConnectionDirection(conn)
+		fixIncomingConnectionDirection(conn)
```

Comment:
nit: For someone who hasn't read the implementation, this _may_ give the impression that the both fixes may be applicable for a single connection (i.e., an outgoing connection may get "fixed" first into incoming and then it can get again "fixed"and become outgoing). We may consider moving the direction check here to be clearer, i.e.,
```go
if c.Direction == INCOMING {
    fixIncomingConnectionDirection(conn)
} else {
    fixOutgoingConnectionDirection(conn)
}
```